core/src/hir/defs.rs

```diff
@@ -19,6 +20,8 @@ pub enum TypeDef<'tcx> {
     OutStruct(&'tcx OutStructDef),
     Opaque(&'tcx OpaqueDef),
     Enum(&'tcx EnumDef),
+    Callback(&'tcx Callback),
```

Comment:
issue: I don't think Callbacks should be a separate instance of TypeDef, I think Callbacks should be a variant of Type

https://docs.rs/diplomat_core/latest/diplomat_core/hir/enum.Type.html

There's no reason for callbacks to be separately defined IMO, they can be mentioned in the method and that ought to be sufficient.

Reply:
basically, TypeDef is for custom named types defined by the user, like `struct Foo {}`

Type is for actual types being used in a struct field, parameter, or return type, and can be a broader set of things.